### ISSUE TITLE
Vagrant cannot ssh

### ISSUE THREAD

**Issue by markdrayton247** [hashicorp/vagrant#11525]
Vagrant cannot ssh to VM on vagrant up. Existing VM and vagrant config has been running fine for months, I restarted and now cannot up anything. No changes I can think of apart from possible OSX updates in background. 

I am able to ssh to the VM from my local OS using a private network IP, so ssh is kinda working, but vagrant cannot provision etc.

### Vagrantfile

```
Vagrant.configure("2") do |config|

  config.vm.define "web-vm" do |webvm|

    webvm.vm.box = "uni-18.04-VM"

    config.vm.provider :vmware_desktop do |vmware|
      vmware.vmx["ethernet0.pcislotnumber"] = "32"
      vmware.gui = true
    end

    webvm.ssh.username = "vagrant"
    webvm.ssh.password = "vagrant"
    webvm.vm.network "forwarded_port", guest: 80, host: 8080
    webvm.vm.network "forwarded_port", guest: 443, host: 8081
    webvm.vm.network "private_network", ip: "192.168.123.100"
    webvm.vm.provision "shell", path: "provision.sh"

  end
```

Errors shown in debug:

```
INFO ssh: Attempting SSH connection...
 INFO ssh: Attempting to connect to SSH...
 INFO ssh:   - Host: 127.0.0.1
 INFO ssh:   - Port: 2222
 INFO ssh:   - Username: vagrant
 INFO ssh:   - Password? true
 INFO ssh:   - Key Path: [redacted]
DEBUG ssh:   - connect_opts: {:auth_methods=>["none", "hostbased", "publickey", "password"], :config=>false, :forward_agent=>false, :send_env=>false, :keys_only=>true, :verify_host_key=>:never, :password=>"vagrant", :port=>2222, :timeout=>15, :user_known_hosts_file=>[], :verbose=>:debug, :encryption=>["aes256-ctr", "aes192-ctr", "aes128-ctr", "aes256-cbc", "aes192-cbc", "aes128-cbc", "rijndael-cbc@lysator.liu.se", "blowfish-ctr", "blowfish-cbc", "cast128-ctr", "cast128-cbc", "3des-ctr", "3des-cbc", "idea-cbc", "arcfour256", "arcfour128", "arcfour", "none"], :logger=>#<Logger:0x0000000101dd2c30 @level=0, @progname=nil, @default_formatter=#<Logger::Formatter:0x0000000101dd2bb8 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x0000000101dd2b68 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<StringIO:0x0000000101dd2c80>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x0000000101dd2b18>>>, :keys=>[redacted], :remote_user=>"vagrant"}
DEBUG ssh: == Net-SSH connection debug-level log START ==
DEBUG ssh: D, [2020-04-17T21:39:19.448325 #3798] DEBUG -- net.ssh.transport.session[80ee804c]: establishing connection to 127.0.0.1:2222
D, [2020-04-17T21:39:19.448657 #3798] DEBUG -- net.ssh.transport.session[80ee804c]: connection established
I, [2020-04-17T21:39:19.448732 #3798]  INFO -- net.ssh.transport.server_version[80eff51c]: negotiating protocol version
D, [2020-04-17T21:39:19.448759 #3798] DEBUG -- net.ssh.transport.server_version[80eff51c]: local is `SSH-2.0-Ruby/Net::SSH_5.1.0 x86_64-darwin19'

DEBUG ssh: == Net-SSH connection debug-level log END ==
 INFO ssh: SSH not up: #<Vagrant::Errors::NetSSHException: An error occurred in the underlying SSH library that Vagrant uses.
The error message is shown below. In many cases, errors from this
library are caused by ssh-agent issues. Try disabling your SSH
agent or removing some keys and try again.

If the problem persists, please report a bug to the net-ssh project.

timeout during server version negotiating>
ERROR warden: Error occurred: Guest-specific operations were attempted on a machine that is not
ready for guest communication. This should not happen and a bug
should be reported.
```

**Comment by briancain** [comment]
:wave: - We chatted on IRC...just to follow up...it might be worth seeing if you can SSH if you don't have a private network on your box. Also, please provide us with the full debug log from running `vagrant up --debug`, as well as `vagrant ssh --debug`. Thanks!

**Comment by markdrayton247** [comment]
Have discovered that an update for VMWare was available, which seems to have fixed the problem, although there is nothing in their release notes that indicates they resolved anything relevant for OSX 10.15 . The issue seems to have been OS related anyway, not vagrant or my box/config.
